src/components/NurseryWithdrawals/NurseryReassignment.tsx

```diff
@@ -1,13 +1,193 @@
+            return acc;
+          }, {})
+        );
+      } else {
+        snackbar.toastError();
+      }
+    };
+
+    populateSpecies();
+  }, [organization.id, snackbar]);
+
+  // populate delivery
+  useEffect(() => {
+    if (!deliveryId) {
+      return;
+    }
+
+    const populateDelivery = async () => {
+      const response = await getDelivery(Number(deliveryId));
+      if (response.requestSucceeded) {
+        setDelivery(response.delivery);
+      } else {
+        snackbar.toastError(response.error);
+      }
+    };
+
+    populateDelivery();
+  }, [deliveryId, snackbar]);
+
+  // populate lookup maps of zones and plots
+  useEffect(() => {
+    if (!delivery) {
+      return;
+    }
+
+    const populatePlots = async () => {
+      const plotIds = delivery.plantings.map((planting) => planting.plotId).filter((id) => id);
+      if (!plotIds.length) {
+        return;
+      }
+      const response = await getPlantingSite(delivery.plantingSiteId);
+      if (response.requestSucceeded && response.site) {
+        setSiteName(response.site.name);
+        const zoneList = response.site.plantingZones?.filter((zone) =>
+          zone.plots?.filter((plot) => plotIds.indexOf(plot.id) !== -1)
+        );
+        if (!zoneList?.length) {
+          return;
+        }
+        const zonesMap: ZonesMap = {};
+        setPlots(
+          zoneList.reduce((acc: any, zone: any) => {
+            const zonePlots = zone.plots;
+            if (zonePlots) {
+              zonePlots.forEach(
+                (plot: Plot) =>
+                  (zonesMap[plot.id.toString()] = { id: zone.id, name: zone.name, plotName: plot.fullName })
+              );
+              acc[zone.id.toString()] = zonePlots.map((plot: Plot) => ({ id: plot.id, fullName: plot.fullName }));
+            }
+            return acc;
+          }, {})
+        );
+        setZones(zonesMap);
+      } else {
+        snackbar.toastError(response.error);
+      }
+    };
+
+    populatePlots();
+  }, [delivery, snackbar]);
+
+  const goToWithdrawals = () => {
+    history.push({ pathname: APP_PATHS.NURSERY_WITHDRAWALS });
+  };
+
+  const reassign = () => {
+    // do nothing for now
+    return;
+  };
+
+  const plantings = useMemo(() => {
+    if (!delivery) {
+      return [];
+    }
+
+    return delivery.plantings
+      .filter((planting) => planting.type === 'Delivery' && planting.plotId)
+      .map((planting) => {
+        if (!planting.plotId || !speciesMap || !zones) {
+          return planting;
+        }
+        return {
+          ...planting,
+          species: speciesMap[planting.speciesId],
+          zone: zones[planting.plotId]?.name,
+          plantingSite: siteName,
+          originalPlot: zones[planting.plotId]?.plotName,
+        };
+      });
+  }, [delivery, siteName, speciesMap, zones]);
+
+  const reassignmentRenderer = useMemo(
+    () =>
+      ReassignmentRenderer({
+        plots: plots || {},
+        zones: zones || {},
```

Comment:
Why do you need the zones in the table?

Reply:
I don't know yet. This is intermediate, I will remove it later if completely not needed.